plugins/woocommerce/includes/class-wc-cart.php

```diff
@@ -1573,10 +1573,26 @@ public function show_shipping() {
 				return false;
 			}
 			$country_fields = WC()->countries->get_address_fields( $country, 'shipping_' );
-			if ( isset( $country_fields['shipping_state'] ) && $country_fields['shipping_state']['required'] && ! $this->get_customer()->get_shipping_state() ) {
+			/**
+			 * Filter to not require shipping state for shipping calculation, even if it is required at checkout.
+			 * This can be used to allow shipping calculations to be done without a state.
+			 *
+			 * @since 8.4.0
+			 *
+			 * @param bool $show_state Whether to use the state field. Default true.
+			 */
+			if ( apply_filters( 'woocommerce_shipping_calculator_enable_state', true ) && isset( $country_fields['shipping_state'] ) && $country_fields['shipping_state']['required'] && ! $this->get_customer()->get_shipping_state() ) {
```

Comment:
```suggestion
			$state_enabled  = apply_filters( 'woocommerce_shipping_calculator_enable_state', true );
			$state_required = isset( $country_fields['shipping_state'] ) && $country_fields['shipping_state']['required'];
			if ( $state_enabled && $state_required && ! $this->get_customer()->get_shipping_state() ) {
```

Not a required change, but I think this would be more readable.

Reply:
Yep, agreed.